x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/EvalMapper.java

```diff
@@ -50,23 +51,33 @@ public final class EvalMapper {
     private EvalMapper() {}
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
-    public static ExpressionEvaluator.Factory toEvaluator(Expression exp, Layout layout) {
+    public static ExpressionEvaluator.Factory toEvaluator(FoldContext foldCtx, Expression exp, Layout layout) {
```

Comment:
Why do we need to pass the fold context to the evaluator? Is it for cases where `toEvaluator` itself needs to perform folding to produce the evaluator?

I've seen e.g. in `Bucket.toEvaluator` that we do a lot of folding. I wonder if that folding is actually needed; maybe the correct way is to rely on the values having been folded already.

In any case, a comment would help clarify that for future us.

Reply:
Yeah - it's for all of the calls to `fold` in the `toEvaluator` implementations. I think for this round we're not going to remove those calls so we should keep this in. But I'd be quite happy to remove this parameter when it isn't required any more.